terraform/custom_domains/environment_domains/main.tf

```diff
@@ -0,0 +1,53 @@
+module "domains" {
+  source              = "git::https://github.com/DFE-Digital/terraform-modules.git//domains/environment_domains?ref=0.5.4"
+  zone                = var.zone
+  front_door_name     = var.front_door_name
+  resource_group_name = var.resource_group_name
+  domains             = var.domains
+  environment         = var.environment_short
+  host_name           = var.origin_hostname # ADDED
+  rule_set_ids        = [azurerm_cdn_frontdoor_rule_set.ruleset.id]
+
+}
+
+data "azurerm_cdn_frontdoor_profile" "main" {
+  name                = var.front_door_name
+  resource_group_name = var.resource_group_name
+}
+
+data "azurerm_dns_zone" "main" {
+  name                = var.zone
+  resource_group_name = var.resource_group_name
+}
+
+resource "azurerm_cdn_frontdoor_rule_set" "ruleset" {
+  name                     = var.environment_short
+  cdn_frontdoor_profile_id = data.azurerm_cdn_frontdoor_profile.main.id
+}
+
+resource "azurerm_cdn_frontdoor_rule" "rule" {
+  for_each   = var.redirect_rules
+  depends_on = [module.domains]
+  name = replace(
+    replace(each.key, ".", ""),
+  "-", "")
+  cdn_frontdoor_rule_set_id = azurerm_cdn_frontdoor_rule_set.ruleset.id
+  order                     = 1
+  behavior_on_match         = "Continue"
+
+  actions {
+
+    url_redirect_action {
+      redirect_type        = "Moved"
+      redirect_protocol    = "Https"
+      destination_hostname = each.value
+    }
+  }
+
+  conditions {
+    host_name_condition {
+      operator     = "Equal"
+      match_values = [each.key]
+    }
+  }
+}
```

Comment:
It looks like we're not actually using this anywhere, `var.redirect_rules` is always set to `{}` - can we remove it? We can always add it back later if we learn that we need it.

Reply:
Removed unneeded resource block and associated variables.